feat: add support for PWA

wei-lee opened the pull request:
https://issues.jboss.org/browse/INTLY-3566

This PR fixes 2 things:

1. Add support for PWA. Turns out we don't have it yet and the app won't work if the device is set to offline.
2. Remove the warning about circular dependencies between the auth service module and the sync service module.

To verify:

1. Run the app using `ionic serve --prod`
2. Login with `test/123` and try view and create some tasks (You can do this on a mobile device as well).
3. Use the network tab to turn offline (or put the app into airplane mode).
4. You should still be able to use the app and continue creating and viewing tasks.
5. Bring the device online again and you should see the data is get synced.

wei-lee commented on angular.json:
```diff
@@ -54,17 +55,19 @@
               "extractCss": true,
               "namedChunks": false,
-              "aot": true,
+              "aot": false,
```
we need to turn off `aot` compilation for this app in production mode. For some reason when using `aot`, the manage tasks view won't work and there is an error print out in the console. Something to do with this: https://github.com/angular/angular/issues/23878

wtrocki commented:
An amazing change! Reviewing now

wtrocki commented:
- `ionic build --prod` is required to generate service worker
- `ionic serve` is not affected and works as it was before 
-  auth works fine

wei-lee commented:
@wtrocki thanks, I will made the changes you recommended.

wei-lee commented:
@wtrocki I addressed your comments. I added a new command to make it simple to run the app as PWA app. Also I used `serve` instead of `ionic serve`. I am not sure how to get `ionic serve` to just serving the content from the `www` directory.

darahayes commented on src/mobile-services.json:
```diff
@@ -1,6 +1,41 @@
-  "namespace": "voyager",
-  "clientId": "voyager-ionic-example",
-  "services": []
+  "clientId": "showcase-app-2",
```
Maybe just revert the changes in this file and we're all good!
